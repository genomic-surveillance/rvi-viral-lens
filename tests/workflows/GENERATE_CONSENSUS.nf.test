// Copyright (C) 2023 Genome Surveillance Unit/Genome Research Ltd.
nextflow_workflow {

    name "Test Workflow GENERATE_CONSENSUS"
    script "workflows/GENERATE_CONSENSUS.nf"
    workflow "GENERATE_CONSENSUS"

    test("Should run without failures") {

        when {
            params {
                meta = [
                    sample_id: "FluH1N1_SARSCoV2_hPIV1_33K_ea_P1",
                    taxid: "2697049",
                    id: "FluH1N1_SARSCoV2_hPIV1_33K_ea_P1.2697049",
                    ref_files: [
                        "${projectDir}/tests/test_data/test_references/2697049.fa",
                        "${projectDir}/tests/test_data/test_references/2697049.fa.amb",
                        "${projectDir}/tests/test_data/test_references/2697049.fa.ann",
                        "${projectDir}/tests/test_data/test_references/2697049.fa.bwt",
                        "${projectDir}/tests/test_data/test_references/2697049.fa.pac",
                        "${projectDir}/tests/test_data/test_references/2697049.fa.sa"
                    ]
                ]
                reads = [
                    "${projectDir}/tests/test_data/test_taxon_reads/FluH1N1_SARSCoV2_hPIV1_33K_ea_P1/FluH1N1_SARSCoV2_hPIV1_33K_ea_P1.2697049.extracted_1.fq", 
                    "${projectDir}/tests/test_data/test_taxon_reads/FluH1N1_SARSCoV2_hPIV1_33K_ea_P1/FluH1N1_SARSCoV2_hPIV1_33K_ea_P1.2697049.extracted_2.fq"
                ]
            }
            workflow {
                """
                // define inputs of the workflow here. Example:
                input[0] = Channel.of(tuple(params.meta, params.reads))
                """
            }

            then {
                assert workflow.success
                assert snapshot(workflow.out).match()
            }
        }
    }
}


