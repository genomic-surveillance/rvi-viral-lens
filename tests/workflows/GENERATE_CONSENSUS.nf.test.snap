{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "FluH1N1_SARSCoV2_hPIV1_33K_ea_P1.2697049",
                            "ref_files": [
                                "2697049.fa:md5,7ca8f77fb20a6bcb1eb0ce45d2504268",
                                "2697049.fa.amb:md5,80f653e5d4e35ada77d0a10eb91e4d2d",
                                "2697049.fa.ann:md5,60df9428892d946d9723a235b7e9346b",
                                "2697049.fa.bwt:md5,88f3909bc7c2f9196bd5b1f1dd5d54a6",
                                "2697049.fa.pac:md5,f631387e31f59214b85a8a80f63155f6",
                                "2697049.fa.sa:md5,5bf62e5242a3b6889cb64a38049c6586"
                            ],
                            "taxid": "2697049",
                            "sample_id": "FluH1N1_SARSCoV2_hPIV1_33K_ea_P1",
                            "bam_file": "FluH1N1_SARSCoV2_hPIV1_33K_ea_P1.2697049.sorted.bam:md5,d89eb0b1974e11136558a1ba9533649f",
                            "mpileup_file": "FluH1N1_SARSCoV2_hPIV1_33K_ea_P1.2697049.mpileup.out:md5,f684647fb3c764179cb9cc8924b19cef"
                        },
                        "FluH1N1_SARSCoV2_hPIV1_33K_ea_P1.2697049.consensus.fa:md5,35582ae1636c2b51a7ee183e8df0bfba"
                    ]
                ],
                "out_ch": [
                    [
                        {
                            "id": "FluH1N1_SARSCoV2_hPIV1_33K_ea_P1.2697049",
                            "ref_files": [
                                "2697049.fa:md5,7ca8f77fb20a6bcb1eb0ce45d2504268",
                                "2697049.fa.amb:md5,80f653e5d4e35ada77d0a10eb91e4d2d",
                                "2697049.fa.ann:md5,60df9428892d946d9723a235b7e9346b",
                                "2697049.fa.bwt:md5,88f3909bc7c2f9196bd5b1f1dd5d54a6",
                                "2697049.fa.pac:md5,f631387e31f59214b85a8a80f63155f6",
                                "2697049.fa.sa:md5,5bf62e5242a3b6889cb64a38049c6586"
                            ],
                            "taxid": "2697049",
                            "sample_id": "FluH1N1_SARSCoV2_hPIV1_33K_ea_P1",
                            "bam_file": "FluH1N1_SARSCoV2_hPIV1_33K_ea_P1.2697049.sorted.bam:md5,d89eb0b1974e11136558a1ba9533649f",
                            "mpileup_file": "FluH1N1_SARSCoV2_hPIV1_33K_ea_P1.2697049.mpileup.out:md5,f684647fb3c764179cb9cc8924b19cef"
                        },
                        "FluH1N1_SARSCoV2_hPIV1_33K_ea_P1.2697049.consensus.fa:md5,35582ae1636c2b51a7ee183e8df0bfba"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.0"
        },
        "timestamp": "2024-09-17T15:22:22.278346727"
    }
}